src/rdata/macros.rs

```diff
@@ -506,6 +506,17 @@ macro_rules! rdata_types {
         /// This enum collects the record data types for all currently
         /// implemented record types.
         #[derive(Clone)]
+        #[cfg_attr(
+            feature = "serde",
+            derive(serde::Serialize),
+            serde(bound(
+                serialize = "
+                    O: AsRef<[u8]> + octseq::serde::SerializeOctets,
```

Comment:
@partim I copied this from elsewhere, but isn't `AsRef<[u8]>` always enough because we can do `octs.as_ref().serialize_octets(serializer)`?

Reply:
You are right. It’s possible that `SerializeOctets` was doing more things in an earlier iteration. Or I was just not thinking straight.

Reply:
Ah, but then you have to do your own `Serialize` impl, no? This way we can just derive it?